android/src/main/java/net/no_mad/tts/TextToSpeechModule.java

```diff
@@ -690,27 +736,29 @@ private void abandonFocus() {
         }
     }
 
-    private void ttsCompleted() {
-        isTtsCompleted = true;
-        lock.lock();
-        condition.signal();
-        lock.unlock();
+    private void ttsCompleted(boolean interrupting) {
```

Comment:
This one should only be called while the lock is active, correct? Maybe add a comment to indicate that.

Reply:
Moved code of several small functions only called by audioDone that should be locked to audioDone in commit [06e82b1](https://github.com/tomtom-international/react-native-tts/pull/2/commits/06e82b1674e521a45c3b0ab4e366a8313d46795d)